File: backend/src/main/java/com/example/demo/model/Student.java
package com.example.demo.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;

// sexual: json['stusexual'] as String, 
//       phone: json['stuphone'] as String, 
//       major: json['stumajor'] as String, 
//       grade: json['stugrade'] as String,
//       isLeader: json['stuisLeader'] as bool?,
//       teamid: json['teamid'] as String?,
//       stuIdCard: json['stuIdCard'] as String?
public class Student {
    @JsonProperty("stuid")
    private String stuid;
    @JsonProperty("stuname")
    private String stuname;
    @JsonProperty("stuemail")
    private String stuemail;
    @JsonProperty("stusexual")
    private String stusexual;
    @JsonProperty("stuphone")
    private String stuphone;
    @JsonProperty("stumajor")
    private String stumajor;
    @JsonProperty("stugrade")
    private String stugrade;
    @JsonProperty("stuisleader")
    private Boolean stuisLeader;
    @JsonProperty("teamid")
    private String teamid;
    @JsonProperty("stuIdCard")
    private String stuIdCard;

    public String getStuID() {
        return stuid;
    }

    public void setStuID(String stuid) {
        this.stuid = stuid;
    }

    public String getStuName() {
        return stuname;
    }

    public void setStuName(String stuname) {
        this.stuname = stuname;
    }

    public String getStuEmail() {
        return stuemail;
    }

    public void setStuEmail(String stuemail) {
        this.stuemail = stuemail;
    }

    public String getStuSex() {
        return stusexual;
    }

    public void setStuSex(String stusexual) {
        this.stusexual = stusexual;
    }

    public String getStuPhone() {
        return stuphone;
    }

    public void setStuPhone(String stuphone) {
        this.stuphone = stuphone;
    }

    public String getStuDepartment() {
        return stumajor;
    }

    public void setStuDepartment(String stumajor) {
        this.stumajor = stumajor;
    }

    public String getStuGrade() {
        return stugrade;
    }

    public void setStuGrade(String stugrade) {
        this.stugrade = stugrade;
    }

    public Boolean getStuRole() {
        return stuisLeader;
    }

    public void setStuRole(Boolean stuisLeader) {
        this.stuisLeader = stuisLeader;
    }

    public String getTeamId() {
        return teamid;
    }

    public void setTeamId(String teamid) {
        this.teamid = teamid;
    }

    public String getStuIdCard() {
        return stuIdCard;
    }

    public void setStuIdCard(String stuIdCard) {
        this.stuIdCard = stuIdCard;
    }

    @Override
    public String toString() {
        return "Student{" +
                "stuid='" + stuid + '\'' +
                ", stuname='" + stuname + '\'' +
                ", stuemail='" + stuemail + '\'' +
                ", stusexual='" + stusexual + '\'' +
                ", stuphone='" + stuphone + '\'' +
                ", stumajor='" + stumajor + '\'' +
                ", stugrade='" + stugrade + '\'' +
                ", stuisLeader=" + stuisLeader +
                ", teamid='" + teamid + '\'' +
                ", stuIdCard='" + stuIdCard + '\'' +
                '}';
    }
}

File: frontend/lib/data/Student.dart
import 'package:db_finalproject/data/User.dart';

class Student extends User{
  String major;
  String grade;
  bool? isLeader;
  String? teamid;
  String? stuIdCard;

  Student({
    required super.id,
    super.passwd,
    required super.name,
    required super.email,
    required super.sexual,
    required super.phone,
    required this.major,
    required this.grade,
    this.isLeader,
    this.teamid,
    this.stuIdCard
  });

  factory Student.fromJson(Map<String,dynamic> json){
    return Student(
      id: json['stuid']??'useraccount',
      name: json['stuname'] as String, 
      email: json['stuemail'] as String, 
      sexual: json['stusexual'] as String, 
      phone: json['stuphone'] as String, 
      major: json['stumajor'] as String, 
      grade: json['stugrade'] as String,
      isLeader: json['stuisLeader'] as bool?,
      teamid: json['teamid'] as String?,
      stuIdCard: json['stuIdCard'] as String?
    );
  }

  @override
  Map<String,dynamic> toJson(){
    return {
      'stuid': id,
      'stupasswd': passwd,
      'stuname':name,
      'stuemail': email,
      'stusexual':sexual,
      'stuphone':phone,
      'stumajor':major,
      'stugrade':grade,
      'stuIdCard':stuIdCard
    };
  }

  @override
  void setField(String field,String value){
    switch (field) {
      case 'stuid':
        id = value;
        break;
      case 'stupasswd':
        passwd = value;
        break;
      case 'stuname':
        name = value;
        break;
      case 'stuemail':
        email = value;
        break;
      case 'stusexual':
        sexual = value;
        break;
      case 'stuphone':
        phone = value;
        break;
      case 'stumajor':
        major = value;
        break;
      case 'stugrade':
        grade = value;
        break;
      default:
        break;
    }
  }

  @override
  String getField(String field){
    switch (field) {
      case 'stuid':
        return id;
      case 'stupasswd':
        return passwd!;
      case 'stuname':
        return name;
      case 'stuemail':
        return email;
      case 'stusexual':
        return sexual;
      case 'stuphone':
        return phone;
      case 'stumajor':
        return major;
      case 'stugrade':
        return grade;
      default:
        return '';
    }
  }
}
